fix(bazzite-hardware-setup): run fixups commands after applying kargs

PVermeer opened the pull request:
**Issue:**
The service `bazzite-hardware-setup` fails when updating kargs and internet access is required (updating metadata).

I'm using a vpn (mullvad) that blocks internet access on boot. In this scenario the fixup files are still written to `/etc/bazzite/fixups/**`.
Fixups files should only be written if kargs are successful.

**Fix:**
Collect commands the same way as the needed kargs and run them after applying kargs.

This won't fix the 'no internet' on boot issue but I think that's out of scope for this issue. It will work eventually when metadata gets updated.

The service could also have a trigger to run when the user session has started and if there is an active internet connection when it has failed on boot. `systemctl reboot` Would be inhibited by the user session.

KyleGospo commented:
I'm a little confused, no part of this service should require internet access.

antheas commented:
I simplified the commit a bit. Hopefully you dont mind. The quirks need to always be touched, even if they were not applicable. I just added them to the end.

antheas commented:
i messed up smh

PVermeer commented:
> I simplified the commit a bit. Hopefully you dont mind. The quirks need to always be touched, even if they were not applicable. I just added them to the end.

No probs, I see you already fixed the reboot :)

PVermeer commented:
@antheas

I still have one question:

Should the writing of the version file etc also happen after running kargs? My intention was to retry on every boot for now. It seems to skip when the version is set.

https://github.com/ublue-os/bazzite/blob/de32d3fbdd45cdca9afca05a152490dea774ed6b/system_files/desktop/shared/usr/libexec/bazzite-hardware-setup#L413-L436

Thanks

PVermeer commented:
I see you already did this :thumbsup:.
